app/cluster-api-provider-sidero/controllers/metalmachine_controller.go

```diff
@@ -67,11 +68,15 @@ func (r *MetalMachineReconciler) Reconcile(req ctrl.Request) (_ ctrl.Result, err
 
 	machine, err := util.GetOwnerMachine(ctx, r.Client, metalMachine.ObjectMeta)
 	if err != nil {
-		return ctrl.Result{}, err
+		r.Log.Error(err, "Failed to get machine")
+
+		return ctrl.Result{RequeueAfter: 20 * time.Second}, nil
 	}
 
 	if machine == nil {
-		return ctrl.Result{}, fmt.Errorf("no ownerref for machine %s", metalMachine.ObjectMeta.Name)
+		r.Log.Info("No ownerref for metalmachine", "metalmachine", metalMachine.ObjectMeta.Name)
```

Comment:
Logger is already initialized as:
```go
logger := r.Log.WithValues("metalmachine", req.NamespacedName)
```
Does it override this variable? Are these variable values different?
If they're different maybe it makes sense to initialize logger with `metalMachine.ObjectMeta.Name` otherwise?